.janitor/dspace.dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM janitortechnology/ubuntu-dev
+
+# Add the project 
+ADD ../ .
+
+WORKDIR /home/user/dspace-angular/
+
+# Add server configuration
+COPY --chown=user:user ./home/user/dspace
+COPY environment.prod.js /home/user/dspace-angular/config/
```

Comment:
This operation is binding all of the configuration settings into the image build.  Would it be possible to allow the user to specify these at launch time instead?

Take a look at the approach that I used for the DSpace 7 compose file.  The environmental overrides are supplied when the compose file is started rather than when the image is built.

https://github.com/DSpace-Labs/DSpace-Docker-Images/tree/master/docker-compose-files/dspace7-compose

Would this be possible within Janitor?

Reply:
In Janitor, the environments are pre-built so that they can be started in 2.1 seconds, so I am not sure this is possible. I will investigate and report back. 

cc @jankeromnes

Reply:
Janitor doesn't support Docker compose (yet). However, environment configuration can be done in three different ways:

- Project-specific configuration files (or ENV variables), included in the project image: useful for default configurations that will be useful for most users, yet these can still be tweaked as much as needed by users once their containers are running
- User-specific configuration files, managed across all the user's containers: these files are generated from [templates](https://github.com/JanitorTechnology/janitor/tree/master/templates/configurations), editable in [Settings](https://janitor.technology/settings/), and deployed in *all* the user's containers (useful to configure common tools, regardless of which project they're working on)
- Ad-hoc commands or scripts run by users in their containers: these scripts can be added to `janitor.json`'s `"scripts"` section for convenience (e.g. the Discourse image has a `bundle exec rake admin:create` script that users can decide to run, either manually or by using the ["Create admin user" IDE script](https://github.com/JanitorTechnology/dockerfiles/blob/9ee71d49efce9f5a7d875c9a82d9f7e9e54d9c19/discourse/janitor.json#L39) in one click)